scripts/leapfrog/elasticsearch_upgrade.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+# Copyright 2017, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+## Shell Opts ----------------------------------------------------------------
+set -e -u -x
+set -o pipefail
+
+# Check upgrade flag
+if [[ -n "$UPGRADE_ELASTICSEARCH" ]]; then
+  if [[ "$UPGRADE_ELASTICSEARCH" == "NO" ]]; then
```

Comment:
The default in the above variable is "no" should the above default option be "NO"?

Also with the casting of strings we might want to consider making this `true or `false`. 

something like:

``` bash
if [ ${UPGRADE_ELASTICSEARCH} = false ]; then
...
fi
```

Reply:
yeah, I was confused with all existing env var we're using. Sometime it is true/false, and sometimes it is yes/no. I'll match the var value in default here, I mean match 'no' to 'NO'.